### PR TITLE
Issue #295 custom gradebook columns

### LINKED ISSUE #295: Custom Gradebook Columns
**What resource needs additional coverage?**

Custom Gradebook Columns

**What endpoints need to be covered?**

https://canvas.instructure.com/doc/api/custom_gradebook_columns.html

Custom Gradebook Columns:
- [ ] List custom gradebook columns
- [ ] Create a custom gradebook column
- [ ] Update a custom gradebook column
- [ ] Delete a custom gradebook column
- [ ] Reorder custom columns

Custom Gradebook Column Data:
- [ ] List entries for a column
- [ ] Update column data
- [ ] Bulk update column data

### PR DESCRIPTION
**Pull request by aileenpongnon**
# Proposed Changes

* Implemented endpoints (create custom gradebooks column, list custom gradebook columns, bulk update column data)  into course.py 
* Implemented endpoints (update custom gradebook column, delete a custom gradebook column, reorder custom columns, list entries for a column, update column data) into custom_gradebook_columns.py
* Updated test_course.py and course.json to include additional unit tests and JSON 
* Created test_custom_gradebook_columns.py and custom_gradebook_columns.json to test endpoints and JSON

Fixes #295

### CODE REVIEW

File: canvasapi/custom_gradebook_columns.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+        response = self._requester.request(
+            "POST",
+            "courses/{}/custom_gradebook_columns/reorder".format(self.course_id),
+            order=order,
+        )
+
+        return response.json().get("reorder")
+
+    def update_custom_column(self, column, **kwargs):
```
[the remark this answers is not in the document]
Comment (aileenpongnon) [reply canvasapi/custom_gradebook_columns.py:79]: It says that it accepts the same parameters as custom gradebook column creation, which requires column[title] ([ref](https://canvas.instructure.com/doc/api/custom_gradebook_columns.html#method.custom_gradebook_columns_api.update))

[the remark this answers is not in the document]
Comment (Thetwam) [reply canvasapi/custom_gradebook_columns.py:79]: I accepts, but does not require them.

File: canvasapi/course.py [hunk 2]
```diff
@@ -216,6 +239,32 @@ def create_course_section(self, **kwargs):
...
 
+    def create_custom_column(self, column, **kwargs):
+        """
+        Create a custom gradebook column.
+
+        :calls: `POST /api/v1/courses/:course_id/custom_gradebook_columns \
+        <https://canvas.instructure.com/doc/api/custom_gradebook_columns.html#method.custom_gradebook_columns_api.create>`_
+
+        param column[title]: The title of the column.
+        :type column[title]: str
```
[the remark this answers is not in the document]
Comment (Thetwam) [reply canvasapi/course.py]: Syntax still broken after change.  Build and view docs to see.

File: canvasapi/custom_gradebook_columns.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    def reorder_custom_columns(self, order):
+        """
+        Put the given columns in a specific order based on given parameter.
+
+        :calls: `POST /api/v1/courses/:course_id/custom_gradebook_columns/reorder \
+            <https://canvas.instructure.com/doc/api/custom_gradebook_columns.html#method.custom_gradebook_columns_api.reorder>`_
+
+        :param order: The order the columns are supposed to be in.
+        :type order: int
```
[the remark this answers is not in the document]
Comment (Thetwam) [reply canvasapi/custom_gradebook_columns.py]: Sorry, should have put this in quotes. Type should literally be "list of int", not list[int]

File: canvasapi/custom_gradebook_columns.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thetwam) [reply canvasapi/custom_gradebook_columns.py:80]: Appears to still be incorrectly required.

File: canvasapi/custom_gradebook_columns.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            super(CustomGradebookColumn, self).set_attributes(response.json())
+
+        return self
+
+
+class ColumnData(CanvasObject):
+    def __str__(self):
+        return "{} ({})".format(self.user_id, self.content)
+
+    def update_column_data(self, **kwargs):
```
Comment (Thetwam) [inline canvasapi/custom_gradebook_columns.py:110]: It looks like `column_data` was accidentally removed from here? Docs say it's required.

File: docs/custom-gradebook-columns-ref.rst [hunk 5]
```diff
@@ -0,0 +1,13 @@
+========================
+Custom Gradebook Column
```
Comment (Thetwam) [inline docs/custom-gradebook-columns-ref.rst:2]: This should be the classname (no spaces)

File: canvasapi/module.py [hunk 6]
```diff
@@ -126,9 +126,10 @@ def create_module_item(self, module_item, **kwargs):
             if "content_id" in module_item:
                 kwargs["module_item"] = module_item
             else:
-                raise RequiredFieldMissing(
-                    "Dictionary with key 'content_id' is required."
-                )
+                if "type" in module_item != ("ExternalUrl" or "Page" or "SubHeader"):
+                    raise RequiredFieldMissing(
+                        "Dictionary with key 'content_id' is required."
+                    )
```
Comment (Thetwam) [inline canvasapi/module.py:132]: What's up with this change?

Comment (aileenpongnon) [reply canvasapi/module.py:132]: an attempt at solving #350, not working though 👎🏾

Comment (Thetwam) [reply canvasapi/module.py:132]: Gonna remove this from this branch/PR.

**Comment by Thetwam** [comment]
Looks good 👍 

Great work, @aileenpongnon!

**Pull request merged**